File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int>ans(nums.size());
        int prefix=1;
        for(int i=0;i<nums.size();i++){
            ans[i]=prefix;
            prefix*=nums[i];
        }
        int  sufix=1;
        for(int j=ans.size()-1;j>=0;j--){
            ans[j]=ans[j]*sufix;
            sufix*=nums[j];
        }
        return ans;
    }
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        sol=[]
        prefix=1
        for i in range(len(nums)):
            sol.append(prefix)
            prefix *= nums[i]
        suffix = 1
        for i in range(len(nums)-1,-1,-1):
            sol[i]*=suffix
            suffix *= nums[i]
        return sol
        